app/services/chat.ts

```diff
@@ -0,0 +1,190 @@
+import { Service } from 'services/service';
+import { Inject } from 'util/injector';
+import { UserService } from 'services/user';
+import { getPlatformService } from 'services/platforms';
+import { CustomizationService, ICustomizationSettings } from 'services/customization';
+import electron from 'electron';
+import { YoutubeService } from 'services/platforms/youtube';
+import url from 'url';
+import { WindowsService } from 'services/windows';
+import { $t } from 'services/i18n';
+
+export class ChatService extends Service {
+  @Inject() userService: UserService;
+  @Inject() customizationService: CustomizationService;
+  @Inject() windowsService: WindowsService;
+
+  private chatView: Electron.BrowserView;
+
+  init() {
+    this.userService.userLogin.subscribe(() => this.initChat());
+    this.userService.userLogout.subscribe(() => this.deinitChat());
+
+    if (this.userService.isLoggedIn()) this.initChat();
+  }
+
+  refreshChat() {
+    this.navigateToChat();
+  }
+
+  mountChat(electronWindowId: number) {
+    if (!this.chatView) return;
+
+    const win = electron.remote.BrowserWindow.fromId(electronWindowId);
+
+    // This method was added in our fork
+    (win as any).addBrowserView(this.chatView);
```

Comment:
Nitpick, prefer:
`// @ts-ignore: this method was added in our fork`
so we don't have to remove (or read) the typecast if we ever fix typings.

Reply:
Did not realize this was a thing.